### DISCUSSION TITLE
useAsyncList prevent initial fetch - provide predicate?

### DISCUSSION THREAD

**Discussion by binaryartifex** [adobe/react-spectrum#4014] [Q&A]
Im following along with this example [here](https://codesandbox.io/s/dreamy-burnell-3i2jy?file=/src/App.tsx) of how to make the humble search field. requirements are simple - no fetching until at least 3 debounced characters are ready to provide as the initial query. No client side filtering necessary as the api response has that handled. The response popover should be navigable via the up and down arrows while permitting further typing, which i belief useComboBox already does. Onpress for any option simply navigates to that particular page. I "think" i can achieve most of this with react-aria, but im stuck on enforcing some type of predicate against the useAsyncList on initial render, as these search items are information rich and an initial search on an empty string is expensive....

**Comment by SteffenFrankSDC** [comment]
You can just add an early return to your `load` callback, that returns an empty result set if the input is less than three characters long:

```js
async load({ signal, cursor, filterText }) {
   if (!filterText || filterText.length < 3) return {items: []};
   // ...
}
```

**Reply by binaryartifex** [reply]
that was my initial thought as well, however you'll find that if your items array is initially empty, you need to set the `allowsEmptyCollection` predicate when you declare the useComboBoxState hook. what happens in this instance is that when your load predicate is met, and the results are fetched, the dropdown doesn't appear, but subsequent characters will. so the search results seem to be out of sync with the dropdown state. I did manage to fix the issue (although a few concerns remain outstanding, within an issue i submitted at #4016)
